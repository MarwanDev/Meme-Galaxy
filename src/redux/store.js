import { configureStore } from "@reduxjs/toolkit";
import memesReducer from "./memesSlice/memesSlice";

const store = configureStore({
  reducer: {
    meme: memesReducer,
  },
});

export default store;
