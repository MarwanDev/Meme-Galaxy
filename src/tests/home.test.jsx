import { Home } from "../pages";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import renderer from "react-test-renderer";
import store from "../redux/store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

describe("Home", () => {
  it("renders each category correctly", () => {
    const home = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Home />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(home).toMatchSnapshot();
  });
  it("renders Category component correctly", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>
    );
  });
});
