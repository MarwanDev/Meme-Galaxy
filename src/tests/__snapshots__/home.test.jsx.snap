// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > renders each category correctly 1`] = `
[
  <nav
    className="navbar-container"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      style={
        {
          "backgroundColor": "#4369b2",
          "color": "#fff",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
    <h1
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#4369b2",
          "color": "#fff",
          "fontSize": 18,
          "margin": "auto",
          "textAlign": "center",
        }
      }
    >
      all memes stat
    </h1>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MicIcon"
      focusable="false"
      style={
        {
          "backgroundColor": "#4369b2",
          "color": "#fff",
          "marginLeft": "auto",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SettingsIcon"
      focusable="false"
      style={
        {
          "backgroundColor": "#4369b2",
          "color": "#fff",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
      />
    </svg>
  </nav>,
  <div
    className="header-container"
  >
    <h2
      style={
        {
          "backgroundColor": "#4369b2",
          "color": "#fff",
          "marginBottom": 15,
          "textAlign": "center",
        }
      }
    >
      memes stat
    </h2>
    <ul
      className="info-container"
    >
      <li
        className="info"
      >
        <h4>
          0
        </h4>
        <h3>
          memes
        </h3>
      </li>
      <li
        className="info"
      >
        <h4>
          -Infinity
        </h4>
        <h3>
          categories
        </h3>
      </li>
      <li
        className="info"
      >
        <h4>
          -Infinity
        </h4>
        <h3>
          authors
        </h3>
      </li>
    </ul>
  </div>,
  <div
    className="categories-container"
  >
    <ul
      style={
        {
          "display": "flex",
          "flexWrap": "wrap",
          "gap": 40,
          "justifyContent": "space-between",
          "listStyleType": "none",
          "paddingTop": 10,
        }
      }
    >
      <li>
        <div
          className="category-container"
          href=""
          id={2}
          onClick={[Function]}
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            style={
              {
                "alignSelf": "flex-end",
                "color": "#fff",
                "textAlign": "end",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <a
            className=""
            href="/details"
            id={2}
            onClick={[Function]}
            style={
              {
                "fontSize": 60,
                "textDecoration": "none",
              }
            }
          >
            ðŸ¤£
          </a>
          <h6
            style={
              {
                "alignSelf": "flex-end",
                "color": "#fff",
                "marginTop": 4,
                "textAlign": "right",
                "textDecoration": "none",
              }
            }
          >
            Mr Bala7
          </h6>
        </div>
      </li>
      <li>
        <div
          className="category-container"
          href=""
          id={3}
          onClick={[Function]}
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            style={
              {
                "alignSelf": "flex-end",
                "color": "#fff",
                "textAlign": "end",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <a
            className=""
            href="/details"
            id={3}
            onClick={[Function]}
            style={
              {
                "fontSize": 60,
                "textDecoration": "none",
              }
            }
          >
            ðŸ¤¡
          </a>
          <h6
            style={
              {
                "alignSelf": "flex-end",
                "color": "#fff",
                "marginTop": 4,
                "textAlign": "right",
                "textDecoration": "none",
              }
            }
          >
            Mr Betengana
          </h6>
        </div>
      </li>
      <li>
        <div
          className="category-container"
          href=""
          id={4}
          onClick={[Function]}
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            style={
              {
                "alignSelf": "flex-end",
                "color": "#fff",
                "textAlign": "end",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <a
            className=""
            href="/details"
            id={4}
            onClick={[Function]}
            style={
              {
                "fontSize": 60,
                "textDecoration": "none",
              }
            }
          >
            ðŸ˜…
          </a>
          <h6
            style={
              {
                "alignSelf": "flex-end",
                "color": "#fff",
                "marginTop": 4,
                "textAlign": "right",
                "textDecoration": "none",
              }
            }
          >
            Mr Basterma
          </h6>
        </div>
      </li>
      <li>
        <div
          className="category-container"
          href=""
          id={5}
          onClick={[Function]}
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            style={
              {
                "alignSelf": "flex-end",
                "color": "#fff",
                "textAlign": "end",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <a
            className=""
            href="/details"
            id={5}
            onClick={[Function]}
            style={
              {
                "fontSize": 60,
                "textDecoration": "none",
              }
            }
          >
            ðŸ˜Ž
          </a>
          <h6
            style={
              {
                "alignSelf": "flex-end",
                "color": "#fff",
                "marginTop": 4,
                "textAlign": "right",
                "textDecoration": "none",
              }
            }
          >
            Mrs Bala7
          </h6>
        </div>
      </li>
      <li>
        <div
          className="category-container"
          href=""
          id={0}
          onClick={[Function]}
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            style={
              {
                "alignSelf": "flex-end",
                "color": "#fff",
                "textAlign": "end",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <a
            className=""
            href="/details"
            id={0}
            onClick={[Function]}
            style={
              {
                "fontSize": 60,
                "textDecoration": "none",
              }
            }
          >
            ðŸ¤‘  
          </a>
          <h6
            style={
              {
                "alignSelf": "flex-end",
                "color": "#fff",
                "marginTop": 4,
                "textAlign": "right",
                "textDecoration": "none",
              }
            }
          >
            All
          </h6>
        </div>
      </li>
    </ul>
  </div>,
]
`;
